src/imitation/data/rollout.py

```diff
@@ -460,9 +493,10 @@ def generate_trajectories(
     for trajectory in trajectories:
         n_steps = len(trajectory.acts)
         # extra 1 for the end
-        exp_obs = (n_steps + 1,) + venv.observation_space.shape
-        real_obs = trajectory.obs.shape
-        assert real_obs == exp_obs, f"expected shape {exp_obs}, got {real_obs}"
+        if not isinstance(venv.observation_space, spaces.dict.Dict):
```

Comment:
probably want some analogous check for dict observation spaces? Dict is a composite space so should be able to do some kind of recursive check I think: https://gymnasium.farama.org/api/spaces/composite/